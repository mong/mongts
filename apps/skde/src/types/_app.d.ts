export {};

declare global {
  interface Window {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    _mtm: any;
  }
}
