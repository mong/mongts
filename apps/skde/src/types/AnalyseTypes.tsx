export type NorwegianOrEnglishText = {
  no: string;
  en?: string;
};

export type AnalyseData = {
  tags: string[];
  published: number;
  graphs: {
    type: string;
    category: string;
    description: NorwegianOrEnglishText;
    variables: string[];
    data_id: string;
  }[];
  data: {
    [dsname: string]: {
      [varname: string]: string | number;
    }[];
  };
};

export type Tag = {
  name: string;
  fullname: string;
  introduction?: string;
};
