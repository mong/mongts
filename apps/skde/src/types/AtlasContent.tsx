export type Tekst = {
  type: "tekst";
  tekst: string;
};

export type Faktaboks = {
  type: "faktaboks";
  overskrift: string;
  tekst: string;
  lang: "nb" | "en" | "nn";
};

export type Resultatboks = {
  type: "resultatboks";
  overskrift: string;
  data: string;
  ingress: string;
  utvalg: string;
  resultat: string;
  lang: "nb" | "en" | "nn";
  publisert: Date;
  oppdatert: Date;
};

export type ChapterProps = {
  overskrift?: string;
  innhold: (Tekst | Faktaboks | Resultatboks)[];
  lang: "nb" | "en" | "nn";
};

export type ChaptersProps = {
  innhold: ChapterProps[];
  lang: "nb" | "en" | "nn";
};
