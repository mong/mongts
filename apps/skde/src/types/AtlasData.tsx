export type AtlasData = {
  innbyggere: number;
  bohf: string;
  antall1: number;
  antall2: number;
  andelRate1: number;
  andelRate2: number;
  rate1: number;
  rate2: number;
  year: number | string;
  rateSnitt: number;
  rate2018: number;
  rate2019: number;
  rate2020: number;
  pasienter: number;
  kontakter: number;
  kont_pr_pas: number;
};
