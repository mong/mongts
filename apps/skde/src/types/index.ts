export type { AtlasData } from "./AtlasData";
export type { AnalyseData, Tag } from "./AnalyseTypes";
