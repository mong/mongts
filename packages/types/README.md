# types

Types shared among the packages (`api`, `skde` and `qmongjs`).
