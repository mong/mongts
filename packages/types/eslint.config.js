export default [
  {
    files: ["**/*.ts", "**/*.tsx"],
  },
];
