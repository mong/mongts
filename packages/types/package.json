{
  "name": "types",
  "version": "0.0.0",
  "main": "./src/index",
  "types": "./src/index",
  "packageManager": "yarn@4.2.1",
  "private": true,
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "7.8.0",
    "@typescript-eslint/parser": "7.8.0"
  }
}
