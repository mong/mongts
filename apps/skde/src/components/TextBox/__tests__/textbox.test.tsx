/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { TextBox } from "..";
import { Atlas } from "../../../types";

test("Standard render", async () => {
  const markdownText = "Dette er en test";
  const { container } = render(
    <TextBox atlas={{ lang: "en" } as Atlas}>{markdownText}</TextBox>,
  );
  expect(container).toMatchSnapshot();
});
test("Longer text", async () => {
  const markdownText =
    "Dette er en test\n\n# Hovedoverskrift\nLitt mer tekst\n## Underoverskrift\n### Dette er en overskrift!\n\n```sas\ndata test;\nset test2;\nvar = 0;\nrun;\n```\n\nOg litt tekst her";
  const { container } = render(
    <TextBox atlas={{ lang: "nb" } as Atlas}>{markdownText}</TextBox>,
  );
  expect(container).toMatchSnapshot();
});

test("Text with image", async () => {
  const markdownText =
    'Dette er en test\n\n# Hovedoverskrift\n\n![Menisk, utvikling i antall inngrep i perioden 2013–2017](https://picsum.photos/id/237/100/100.jpg "Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.")\n\n';
  const { container } = render(
    <TextBox atlas={{ lang: "nn" } as Atlas}>{markdownText}</TextBox>,
  );
  expect(container).toMatchSnapshot();
});

test("Text with image, english", async () => {
  const markdownText =
    'Dette er en test\n\n# Hovedoverskrift\n\n![Menisk, utvikling i antall inngrep i perioden 2013–2017](https://picsum.photos/id/237/100/100.jpg "Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.")\n\n';
  const { container } = render(
    <TextBox atlas={{ lang: "en" } as Atlas}>{markdownText}</TextBox>,
  );
  expect(container).toMatchSnapshot();
});
