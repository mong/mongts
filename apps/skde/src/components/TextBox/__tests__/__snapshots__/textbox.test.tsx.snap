// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Longer text 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <p>
    Litt mer tekst
  </p>
  

  <h2
    id="underoverskrift"
  >
    Underoverskrift
  </h2>
  

  <h3
    id="dette-er-en-overskrift"
  >
    Dette er en overskrift!
  </h3>
  

  <pre>
    <code
      class="language-sas"
    >
      data test;
set test2;
var = 0;
run;

    </code>
  </pre>
  

  <p>
    Og litt tekst her
  </p>
</div>
`;

exports[`Standard render 1`] = `
<div>
  <p>
    Dette er en test
  </p>
</div>
`;

exports[`Text with image 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <figure
    style="width: 90%; display: flex; justify-content: center; align-items: center; flex-direction: column;"
  >
    <img
      alt="Menisk, utvikling i antall inngrep i perioden 2013–2017"
      src="https://picsum.photos/id/237/100/100.jpg"
      style="display: block; margin: auto;"
      title="Menisk, utvikling i antall inngrep i perioden 2013–2017"
    />
    <figcaption>
      <strong>
        Figur:
      </strong>
       
      Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.
    </figcaption>
  </figure>
</div>
`;

exports[`Text with image, english 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <figure
    style="width: 90%; display: flex; justify-content: center; align-items: center; flex-direction: column;"
  >
    <img
      alt="Menisk, utvikling i antall inngrep i perioden 2013–2017"
      src="https://picsum.photos/id/237/100/100.jpg"
      style="display: block; margin: auto;"
      title="Menisk, utvikling i antall inngrep i perioden 2013–2017"
    />
    <figcaption>
      <strong>
        Figure:
      </strong>
       
      Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.
    </figcaption>
  </figure>
</div>
`;
